src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -329,6 +330,30 @@ private Blob move() {
       LOG.debug("Successfully deleted {}.", toPath(sourceBlob.getBlobId()));
       return moved;
     }
+  }
+
+  /**
+   * Creates a done file marker (empty file) in the given GCS path.
+   */
+  public void createFileMarker(GCSPath path) {
+    Bucket bucket = null;
+    String bucketName = path.getBucket();
+    String objectName = path.getName();
+    try {
+      bucket = storage.get(bucketName);
+    } catch (StorageException e) {
+
+      throw new RuntimeException(
+        String.format("Unable to access bucket %s. ", bucketName)
+          + "Ensure you entered the correct bucket path.", e);
+    }
+    if (bucket == null) {
+      throw new IllegalArgumentException(
+        String.format("Bucket '%s' does not exist.", bucketName));
```

Comment:
Usually if bucket does not exist we create it. we should do the same here

Reply:
Functionality added to create the bucket if it doesn't exist